### ISSUE TITLE
using setROI returns data not ready. Usage?

### ISSUE THREAD

**Issue by cnieves1** [melopero/Melopero_VL53L1X#3]
Hi, I'm trying to get the measurements for different ROIs using the code:
```

import melopero_vl53l1x as mp

sensor = mp.VL53L1X()
sensor.set_intermeasurement_period(15)
sensor.set_measurement_timing_budget(10)
# Creating 16 ROI definition
i = 0
roiDefs= [[0 for x in range(4)] for y in range(16)]
for y in range(4):
        for x in range(4):
                roiDefs[i][0] = 4*x
                roiDefs[i][1] = (15-4*y)
                roiDefs[i][2] = (4*x+3)
                roiDefs[i][3] = (15-4*y-3)
                i=i+1
measurements =[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
for i in range(10000):
        for  roi in range(16):
                #sensor.stop_ranging()
                sensor.setROI(roiDefs[i][0],roiDefs[i][1],roiDefs[i][2],roiDefs[$
                sensor.start_ranging(mp.VL53L1X.SHORT_DST_MODE)
                measurements[roi] = sensor.get_measurement()
                print(measurements)
                time.sleep(0.1)

sensor.stop_ranging()
sensor.close_connection()
```

The output is always:
 ```
 data not ready
  cause: -7
```

What should I do to use this feature correctly? Thanks!

**Comment by leoli51** [comment]
Hi thank you for the feedback! We will release an update soon with better exceptions management and clearer error strings that will actually explain the error. In the meantime you can try the following: 
-check that the connection are all correct and working  (try to type this line in the terminal: i2cdetect -y 1 . Are you  using the correct i2c address? )
-The minimum and maximum timing budgets are [20 ms, 1000 ms] is your timing budget within this range?  ( -> sensor.set_measurement_timing_budget(10), try changing it to 20 or more)

The error code -7 stands for : " VL53L1_ERROR_TIME_OUT Ranging is aborted due to timeout".
We will test your code in order to find a solution to your problem. 
Sorry for the late response we will reach out to you as soon as possible.

**Comment by leoli51** [comment]
Hi, I tested your code and there is a problem with :
``sensor.set_intermeasurement_period(15)
sensor.set_measurement_timing_budget(10)``

If you comment these two lines the code works. Thank you for pointing out the issue, i will correct this behaviour and update the module in the next few days.

**Comment by leoli51** [comment]
Hi, the problem was that the intermeasurement period must be greater than timing budget + 4 ms...
I released version 0.2.3 which handles this issue and has a way better exception support. I tested your code with the new version after some minor changes and everything works fine (except for an indexError), here is the modified version:

```
import time
import melopero_vl53l1x as mp

sensor = mp.VL53L1X()

sensor.set_measurement_timing_budget(66)
sensor.set_intermeasurement_period(70)

# Creating 16 ROI definition
i = 0
roiDefs= [[0 for x in range(4)] for y in range(16)]
for y in range(4):
        for x in range(4):
                roiDefs[i][0] = 4*x
                roiDefs[i][1] = (15-4*y)
                roiDefs[i][2] = (4*x+3)
                roiDefs[i][3] = (15-4*y-3)
                i=i+1
measurements =[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
sensor.start_ranging(mp.VL53L1X.SHORT_DST_MODE) #should call this only once 

for i in range(10000):
        for  roi in range(16):
                sensor.setROI(roiDefs[i][0],roiDefs[i][1],roiDefs[i][2],roiDefs[i][3])
                measurements[roi] = sensor.get_measurement()
                print(measurements)
                time.sleep(0.1)

sensor.stop_ranging()
sensor.close_connection()```

**Comment by cnieves1** [comment]
Hi,
the problem was not exactly that...  

The intermeasurement period (15) was already greater than timing budget (10) + 4 ms .
The problem was that I was setting intermeasurement period before setting the timing budget.

The new version has better error report, so I noticed that, and setting the timing budget first and the intermeasurement period afterwards (like you did in your example) worked.

The problem is now solved. Thanks for your help!
